SeQuant/domain/mbpt/spin.cpp

```diff
@@ -1167,7 +1167,7 @@ container::svector<ResultExpr> closed_shell_spintrace(
       is_direct_full_expansion);
 }
 
-ExprPtr closed_shell_CC_spintrace_compact_set(ExprPtr const& expr) {
+ExprPtr closed_shell_CC_spintrace_optimal(ExprPtr const& expr) {
```

Comment:
it seems we have settled on the pattern with `spintrace` appearing at the end of the function name, so let's keep it. Here's what I propose:
- rename `closed_shell_CC_spintrace` to `closed_shell_CC_v0_spintrace`
- rename `closed_shell_CC_spintrace_compact_set` to `closed_shell_CC_v1_spintrace`
- re-introduce `closed_shell_CC_spintrace` and have it dispatch to to `closed_shell_CC_v1_spintrace` by default.
- update dox. It's probably better to refer to new equations as v1  rather than compact/optimal (not sure how to prove optimality, we may discover yet better way to do things in the future ...)